### PR TITLE
Update `cut-a-new-release.md`

### PR DESCRIPTION
**Pull request by lougeniaC64**
Most of the changes in this PR are minor edits to clarify the release cutting process. However, a few changes are noteworthy and open for discussion:
- I have moved the initial smoke and sync sample app tests from the release cutting process to the pull request checklist as I think the onus of identifying and managing breaking changes belongs with the creator of the changes, not with whoever happens to be cutting a release at any given time.
- Softened the guidance around running the sample app from local changes. In order for this step to be consistently meaningful as an end-to-end test we should have explicit instructions for how to configure the app (i.e. what emulators to use) and what steps should be performed once the app is running.
- Added guidance in the pull request checklist that a new release should be cut after you've merged breaking changes
 
### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: docs/howtos/cut-a-new-release.md
Comment (lougeniaC64) [inline docs/howtos/cut-a-new-release.md:4]: I'm working on getting the smoke tests to pass on a local branch with nothing but these doc changes. If I can't manage to do that, I'm going to remove the smoke tests from this step as I don't think it would be fair for someone (particularly the uninitiated) to resolve issues running these tests that are unrelated to any breaking changes.

File: docs/howtos/cut-a-new-release.md
Comment (jdragojevic) [inline docs/howtos/cut-a-new-release.md:12]: Are there any instructions for how to run the app and what to do with it?  Is this necessary if we have the other tests running?

File: docs/howtos/cut-a-new-release.md
Comment (jdragojevic) [inline docs/howtos/cut-a-new-release.md:15]: I was confused at this step previously.  

- We don't mention updating the changelog file from unreleased changes.
- Does the PR have to be reviewed and merged and ci completed before moving to step 3?

**Comment by jdragojevic** [comment]
A few other places where I was confused...

- Step 3 says to draft a new release - but it doesn't say to actually Publish it once you've put in all the info.
- Step 4 - should maybe be a note after the fact?  Since I think we're hoping a manual release won't be necessary.
- Step 5 - Maybe it should be the follow-up steps for Android and iOS.  ie - iOS ci ([bitrise](https://app.bitrise.io/app/6c06d3a40422d10f#/builds)) will detect the new release and test it automatically.  Fenix ci will detect and update to  minor releases.  Are there other communication steps to our consumers, or follow-ups to make sure there's no breakage?

File: docs/howtos/cut-a-new-release.md
Comment (lougeniaC64) [reply docs/howtos/cut-a-new-release.md:12]: For posterity, I've moved the smoke test steps to the pull request template as the responsibility of ensuring that breaking changes are identified and addressed should belong to the person who created the changes rather than the person cutting a release (particularly if the release is urgent).

Running the sync sample app (which is currently used by android components to validate app services releases) serves as an end-to-end test. The smoke tests are generally app services related unit tests in consumer repos. So in theory running the sync sample app provides value. In practice, this step is too hand-wavy to provide consistently meaningful value.

Also I highly doubt this is a step that's always executed or that, when it is executed, it provides enough value to warrant the effort. At minimum, if we decide to keep this step, we should explicitly state how the app should be configured (particularly which emulators to use) and what should be tested once the app is running.

File: docs/howtos/cut-a-new-release.md
Comment (mhammond) [inline docs/howtos/cut-a-new-release.md:8]: I struggled to get my head around this - I suggest simplifying it to something like: "Because the release commit must always be on the `main` branch, your PR will need to be approved, CI successful, and merged before a release can be cut.

File: docs/howtos/cut-a-new-release.md
Comment (mhammond) [inline docs/howtos/cut-a-new-release.md:23]: as above :)

**Review by mhammond — APPROVED** [review]
I had a quick look at this when making a release yesterday. r=me and we might as well get it in ASAP, although I think we should say something about the iOS "issue" :)

File: .github/pull_request_template.md
Comment (bendk) [reply .github/pull_request_template.md:5]: The old script ran basically all of the `mode` values except `changes`.  It would be pretty easy to add a an `all-tests` mode that did something very similar.

File: docs/howtos/cut-a-new-release.md
Comment (lougeniaC64) [reply docs/howtos/cut-a-new-release.md:5]: The script should have been updating the changelog file. I've created #4138 to hopefully address this.

File: .github/pull_request_template.md
Comment (lougeniaC64) [reply .github/pull_request_template.md:5]: Okay, I'm going to remove this line. The only reason it's here is to short-circuit the CI feedback loop. If it's more convenient to wait on CI then there's no added value here.

File: docs/howtos/cut-a-new-release.md
Comment (lougeniaC64) [reply docs/howtos/cut-a-new-release.md:23]: Oh, I removed the smoke tests from step 1, and explicitly added them below.

File: .github/pull_request_template.md
Comment (lougeniaC64) [reply .github/pull_request_template.md:8]: Oh man . . . yeah CI doesn't run any of these scripts. I'm going to add a note here and an agenda item for our next kitchen sync. Code committers should have some means to verify code for consumers but this may not be it.

**Review by mhammond — APPROVED** [review]
Thanks - this is definitely an improvement!

**Pull request merged**
